CommonClient.py

```diff
@@ -173,10 +174,75 @@ class CommonContext:
     items_handling: typing.Optional[int] = None
     want_slot_data: bool = True  # should slot_data be retrieved via Connect
 
-    # data package
-    # Contents in flux until connection to server is made, to download correct data for this multiworld.
-    item_names: typing.Dict[int, str] = Utils.KeyedDefaultDict(lambda code: f'Unknown item (ID:{code})')
-    location_names: typing.Dict[int, str] = Utils.KeyedDefaultDict(lambda code: f'Unknown location (ID:{code})')
+    class NameLookupDict(collections.abc.MutableMapping):
```

Comment:
Is there a reason we need this to be a subclass of `collections.abc.MutableMapping`?
It looks like we wouldn't want someone to mutate this through `__setitem__`; it should only be mutated through `update_game`, right?
(and `__delitem__` would also make inconsistencies in `_flat_store` and `_archipelago_lookup`)

Reply:
I used mutable mapping because the instance itself doesn't hold the data, it's just an interface. That's the main reason it's not just a subclass of `dict`.

As for `__setitem__` and `__delitem__`, I suppose those could just raise an exception, but technically before others could modify the item_names and location_names too, so I just implemented that as is. 

Also I don't care if deleting an item messes with `_flat_store`, but `_archipelago_lookup` could be problematic. Maybe I should just change them both to raise instead. I'll sleep on it.

Reply:
And I'm on the fence about whether it should even be a subclass of `Mapping`
We might inherit methods doing strange things that we haven't thought of.
If we are going to inherit stuff, I'd feel more comfortable with unit tests for each of the inherited methods, just to make sure they're doing something sane.

Reply:
> I used mutable mapping because the instance itself doesn't hold the data, it's just an interface. That's the main reason it's not just a subclass of `dict`.
> 
> As for `__setitem__` and `__delitem__`, I suppose those could just raise an exception, but technically before others could modify the item_names and location_names too, so I just implemented that as is.
> 
> Also I don't care if deleting an item messes with `_flat_store`, but `_archipelago_lookup` could be problematic. Maybe I should just change them both to raise instead. I'll sleep on it.

But if it's an interface that we don't want, then why use it?
I think it would be better for the methods to not exist at all rather than existing just to raise an exception.

Reply:
Yeah, I just ended up removing them.